### DISCUSSION TITLE
Quorum queue consumer count incorrect after node failure

### DISCUSSION THREAD

**Discussion by lpn-ch** [rabbitmq/rabbitmq-server#2420] [Other] [answered]
Environment:
Windows Server 2019
RabbitMQ version 3.8.5
Erlang version 22.3
5 node cluster
ha-mode all, ha-sync-mode automatic

Initial situation: one queue on node A with one consumer on queue.

Problem: if node A is disconnected from the network the queue master is changed to another node but the queue count stays at 1 when in reality there are no more consumers. The management web UI  shows a count of 1 consumer for the queue but when you click on the queue for details there are no consumers.

The problem does not exist with Classic queues.

**Comment by lukebakken** [comment]
Please remove the `ha-mode` and `ha-sync-mode` parameters and re-test. They do not apply to quorum queues at all. They should not have an effect on quorum queues but just to be certain, please do that.

**Comment by michaelklishin** [comment]
and `"ha-mode": "all"` is [completely unnecessary](https://www.rabbitmq.com/ha.html#replication-factor).

This could be something about how quorum queues keep track of consumers or emit stats. It matters a great deal here what node the consumer is connected to. @lpn-ch can you please clarify that or provide a reasonably detailed set of specific steps we can use to reproduce?

**Comment by michaelklishin** [comment, marked as answer]
I can reproduce this with some nuances worth mentining. Here's an initial state of a three node cluter with a QQ:

<img width="949" alt="01 qq 1 with a consumer" src="https://user-images.githubusercontent.com/1090/89114814-300d2f80-d489-11ea-9501-6f596cb0e521.png">

And a consumer which is connected to node A:

<img width="1326" alt="02 qq 1 consumer connection" src="https://user-images.githubusercontent.com/1090/89114825-461af000-d489-11ea-8521-f1cf9d2a86f2.png">

I then stop node C. The leader moves to node A:

<img width="1285" alt="03 qq without node C" src="https://user-images.githubusercontent.com/1090/89114844-75316180-d489-11ea-8fa8-6101d04de08a.png">

I add one more consumer to node B:

<img width="1325" alt="04" src="https://user-images.githubusercontent.com/1090/89114847-7f536000-d489-11ea-8169-0d5beed630a9.png">

Then I start node C back online:

<img width="1298" alt="05" src="https://user-images.githubusercontent.com/1090/89114849-87130480-d489-11ea-8bfe-d0926b5f5605.png">

and shut down node A:

<img width="1298" alt="07" src="https://user-images.githubusercontent.com/1090/89114854-92fec680-d489-11ea-9482-3d93316d0e47.png">
<img width="1298" alt="06" src="https://user-images.githubusercontent.com/1090/89114857-942ff380-d489-11ea-9125-78c0247f556f.png">

The individual queue page reports the expected consumers on the list but the consumer count is indeed off on both pages:

<img width="1298" alt="08" src="https://user-images.githubusercontent.com/1090/89114860-998d3e00-d489-11ea-8b9a-548f9f0a9162.png">

**Comment by michaelklishin** [comment]
When I bring back node A up, its connection recovers and the state of the system is what is expected again.

**Comment by michaelklishin** [comment]
I will file a new issue about this.

**Comment by lpn-ch** [comment]
Sorry for the delay, I was out for the weekend. Here is how to reproduce my specific issue if it can help.

As requested, I removed the HA polices. I created a three node cluster.

Here is the initial state: one queue with one consumer on node vs-tw-ws2019-p1

![image](https://user-images.githubusercontent.com/25384886/89207986-e2451400-d5bb-11ea-93be-aabb1d68ad14.png)

![image](https://user-images.githubusercontent.com/25384886/89208139-1c161a80-d5bc-11ea-97ae-0272b68abf44.png)

![image](https://user-images.githubusercontent.com/25384886/89208151-20423800-d5bc-11ea-943d-53f9856bf850.png)

I then STOP virtual machine vs-tw-ws2019-p1 (not shutdown) which is the queue master. This is what I get after:

![image](https://user-images.githubusercontent.com/25384886/89208236-436ce780-d5bc-11ea-8cb1-ebbd500c6f4b.png)

Consumer count is still 1

![image](https://user-images.githubusercontent.com/25384886/89208272-4d8ee600-d5bc-11ea-85c2-264a0319899f.png)

But in fact there are no consumers.

![image](https://user-images.githubusercontent.com/25384886/89208286-567fb780-d5bc-11ea-97e6-689ab23870e8.png)

**Comment by michaelklishin** [comment]
See https://github.com/rabbitmq/rabbitmq-server/issues/2421.
